code/modules/maps/template_types/random_exoplanet/random_planet_subtemplates.dm

```diff
@@ -0,0 +1,30 @@
+///Picks all ruins and tries to spawn them on the levels that make up the planet.
+/datum/map_template/planetoid/proc/generate_features(var/datum/planetoid_data/gen_data, var/list/created_level_data, var/whitelist = ruin_tags_whitelist, var/blacklist = ruin_tags_blacklist)
+	var/my_budget = gen_data._budget_override || subtemplate_budget
+	if(my_budget <= 0)
+		return
+
+	var/list/possible_subtemplates = list()
+	var/list/ruins = SSmapping.get_templates_by_category(ruin_category)
+	for(var/ruin_name in ruins)
+		var/datum/map_template/ruin = ruins[ruin_name]
+		var/ruin_tags = ruin.get_ruin_tags()
+		if(whitelist && !(whitelist & ruin_tags))
+			continue
+		if(blacklist & ruin_tags)
+			continue
+		possible_subtemplates += ruin
+
+	if(!length(possible_subtemplates))
+		return //If we don't have any ruins, don't bother
+
+	for(var/datum/level_data/planetoid/LD in created_level_data)
+		if(prob(25))
```

Comment:
Maybe check that we do have other zlevels here before skipping? Otherwise there's 25% chance of a planet without any ruins placed

Reply:
That's a fair point. But I don't really know what's the expectation on exoplanet ruins. Like what's the minimum?

Reply:
Non zero at least, I don't see the need for this check at all tbh. Empty planets are boring to visit.

Reply:
The check is so ruins aren't placed only on the surface level.

Reply:
It doesn't check if it's on surface or not, there's still 25% to skip the lower zlevel too, or skip entirely if it's 1z planet
There's some sort of budget splitting right after the skip too, does that not make it so ruins aren't placed only on surface?

Also pretty much all ruins are mapped with being surface in mind, I think their distribution should be addressed later with some sort of tag setup, don't need to force them to be underground this hard

Reply:
Every single ruin except the drill site and *maybe*  the radioactive anomaly shrine and the monolith expects to be placed on the surface, having either been built there or crashed on the surface.

Reply:
Yeah, I changed it a few hours ago to just put all ruins onto the surface for now.